src/app/users/users.component.ts

```diff
@@ -1,19 +1,64 @@
 import { Component, OnInit } from '@angular/core';
-import { DataService } from '../shared/data-service';
+import { Router } from '@angular/router';
 import { IUser } from './user.interface';
+import { MatDialog } from '@angular/material/dialog';
+import {
+  animate,
+  state,
+  style,
+  transition,
+  trigger,
+} from '@angular/animations';
+import { UserService } from './user.service';
 
 @Component({
   selector: 'app-users',
   templateUrl: './users.component.html',
   styleUrls: ['./users.component.scss'],
+  animations: [
+    trigger('EnterLeave', [
+      state('flyIn', style({ transform: 'translateX(0)' })),
+      transition(':enter', [
+        style({ transform: 'translateX(-100%)' }),
+        animate('0.5s 300ms ease-in'),
+      ]),
+      transition(':leave', [
+        animate('0.3s ease-out', style({ transform: 'translateX(100%)' })),
+      ]),
+    ]),
+  ],
 })
 export class UsersComponent implements OnInit {
-  constructor(private dataService: DataService) {}
+  constructor(
+    private userService: UserService,
+    private router: Router,
+    public dialog: MatDialog
+  ) {}
+
   users!: IUser[];
+  displayedColumns: string[] = ['name', 'email', 'actions'];
 
   ngOnInit(): void {
-    this.dataService.fetchUsers().subscribe((users: IUser[]) => {
+    this.userService.fetchUsers().subscribe((users: IUser[]) => {
       this.users = users;
     });
   }
+  onEdit(id: number) {
+    this.router.navigate(['users', id, 'edit']);
+  }
+
+  openDeleteDialog(id: number) {
+    const dialogRef = this.dialog.open(UserDeleteConfirm);
+
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result) {
+        this.userService.removeUser(id).subscribe();
```

Comment:
remove the subscription.

Reply:
is not necessary to create a subscription mate.

Reply:
Always subscribe!

An HttpClient method does not begin its HTTP request until you call subscribe() on the observable returned by that method. This is true for all HttpClient methods.